pkg/odo/cli/init/params/interactive.go

```diff
@@ -0,0 +1,96 @@
+package params
+
+import (
+	"fmt"
+
+	"github.com/redhat-developer/odo/pkg/catalog"
+	"github.com/redhat-developer/odo/pkg/odo/cli/init/asker"
+)
+
+const (
+	STATE_ASK_LANG = iota
+	STATE_ASK_TYPE
+	STATE_ASK_STARTER
+	STATE_END
+)
+
+// InteractiveBuilder is a backend that will ask init parameters interactively
+type InteractiveBuilder struct {
+	asker         asker.Asker
+	catalogClient catalog.Client
+}
+
+func NewInteractiveBuilder(asker asker.Asker, catalogClient catalog.Client) *InteractiveBuilder {
+	return &InteractiveBuilder{
+		asker:         asker,
+		catalogClient: catalogClient,
+	}
+}
+
+func (o *InteractiveBuilder) IsAdequate(flags map[string]string) bool {
+	return len(flags) == 0
+}
+
+func (o *InteractiveBuilder) ParamsBuild() (InitParams, error) {
+	result := InitParams{}
+	devfileEntries, _ := o.catalogClient.ListDevfileComponents("")
+
+	langs := devfileEntries.GetLanguages()
```

Comment:
This is a good point, open to discussion. Another point is that it is not possible to mock `devfileEntries` in this function.

But I'm not convinced by creating a package-level function, or even a method in the catalog.Client interface, because `GetLanguages` is very specific to `DevfileComponentTypeList`, and I think it better to restrict its visibility to this structure only.

Reply:
> because `GetLanguages` is very specific to `DevfileComponentTypeList`

I agree with you. In my understanding, methods are better for modifying the state of the type, and functions otherwise; which means, in this case,  a function could be better because we aren't changing the state. But your point is right, let's keep it this way for `DevfileComponentTypeList`. 👍🏾